flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -728,6 +731,7 @@ private ChangelogSnapshotState completeRestore(
                 materializationId = Math.max(materializationId, h.getMaterializationID());
             }
         }
+        this.lastFailedMaterializationId = materializationId;
```

Comment:
Sorry, this is my mistake. I originally wanted to set `lastConfirmedMaterializationId` to "restore" `materializationId` so that the first materialization can be triggered. (Or we can add a field to save the id of the first materialization, and check whether it is the first materialization when `initMaterialization`.)